prExplorer/client-app/src/app/stores/productStore.ts

```diff
@@ -0,0 +1,115 @@
+import { makeAutoObservable, runInAction } from "mobx";
+import { Product } from "../models/product";
+import agent from "../api/agent";
+import { v4 as uuid } from "uuid";
+
+export default class ProductStore {
+  productRegistry = new Map<string, Product>();
+  selectedProduct: Product | undefined = undefined;
+  editMode = false;
+  loading = false;
+  loadingInitial = true;
+
+  constructor() {
+    makeAutoObservable(this);
+  }
+
+  get productsSortedByName() {
+    const productList = Array.from(this.productRegistry.values());
+    productList.sort((a, b) => {
+      const titleA = a.title.toLowerCase();
+      const titleB = b.title.toLowerCase();
+      return titleA.localeCompare(titleB);
+    });
+    return productList;
+  }
+
+
+  loadProducts = async () => {
+    try {
+      const products = await agent.Products.list();
+      products.forEach((product) => {
+        this.productRegistry.set(product.id, product);
+      });
+      this.setLoadingInitial(false);
+    } catch (error) {
+      console.log(error);
```

Comment:
It would be nice to think about "nice" errors handling. Normally, when an unexpected error happens, the user should see some nice information about that. `console.log` should never be present in the production code. If you want to log the errors' details somewhere, `console.log` is not the best for that

Reply:
i will try to implement custom error handling with toastr